src/main/database_manager.cpp

```diff
@@ -87,6 +87,29 @@ const string &DatabaseManager::GetDefaultDatabase(ClientContext &context) {
 	return default_entry.catalog;
 }
 
+void DatabaseManager::SetDefaultDatabase(ClientContext &context, const string &new_value) {
+	auto has_active_transaction = context.transaction.HasActiveTransaction();
+	if (!has_active_transaction) {
+		context.transaction.BeginTransaction();
+	}
+
+	auto db_entry = GetDatabase(context, new_value);
+
+	if (!has_active_transaction) {
+		context.transaction.ClearTransaction();
+	}
```

Comment:
I want to validate here that the database referred to by `new_value` actually exists. However, `GetDatabase` requires a transaction to be active. What is the best way to create a transaction when non is active? Or is there another way to validate that the database exists which does not require to start a transaction?